Doc/library/inspect.rst

```diff
@@ -602,7 +602,13 @@ function.
    .. attribute:: Signature.parameters
 
       An ordered mapping of parameters' names to the corresponding
-      :class:`Parameter` objects.
+      :class:`Parameter` objects.  Parameters appear in strict definition
+      order, including keyword-only parameters.
+
+      .. versionchanged:: 3.6
```

Comment:
3.7?

Reply:
I was struck by this, too. But after thinking about it, I think Larry's relying on 3.6 dicts being ordered. I'm relying on the same thing in my 3.6 dataclasses backport.  There are no 3.6 implementations that don't have ordered dicts (since CPython is the only 3.6 implementation in the known universe, and if any others show up, they're likely to have ordered dicts, too). See https://mail.python.org/pipermail/python-dev/2017-December/151325.html for a brief analysis.

Reply:
Duh, you're right, I'll fix.

Eric: I'm not relying on 3.6 dicts being ordered.  getfullargspec represents kwonly parameters in a list, inspect.Signature represents all parameters in an OrderedDict.

Reply:
Larry, btw, maybe we should switch `inspect.Signature` to use plain dicts (since they are ordered in 3.7)?

Reply:
D'oh.

Reply:
IIUC the semantics of OrderedDict and order-preserving dicts are sliiiiiiightly different; at the very least OrderedDict has at least one extra method dict doesn't (OrderedDict.move_to_end()).  On the one hand *probably* nobody would care, but on the other hand I'm not sure making the change would benefit anybody, either.

Reply:
> IIUC the semantics of OrderedDict and order-preserving dicts are sliiiiiiightly different; at the very least OrderedDict has at least one extra method dict doesn't (OrderedDict.move_to_end()).

yep.

> On the one hand probably nobody would care, but on the other hand I'm not sure making the change would benefit anybody, either.

`dict` is faster and more compact.  And since, as you pointed out, `OrderedDict.move_to_end` is highly unlikely to be used in this context, I think it's an OK change for 3.7.  But up to you.

Reply:
I don't think people using inspect.signature are very performance-sensitive.  I think it's best to leave it alone.  (Though I don't know why it's up to me... I was barely involved in inspect.signature!)